web/src/components/button/DownloadVideoButton.tsx

```diff
@@ -0,0 +1,75 @@
+import { useState } from "react";
+import { Button } from "@/components/ui/button";
+import { toast } from "sonner";
+import ActivityIndicator from "../indicators/activity-indicator";
+import { FaDownload } from "react-icons/fa";
+import { formatUnixTimestampToDateTime } from "@/utils/dateUtil";
+
+type DownloadVideoButtonProps = {
+  source: string;
+  camera: string;
+  startTime: number;
+};
+
+export function DownloadVideoButton({
+  source,
+  camera,
+  startTime,
+}: DownloadVideoButtonProps) {
+  const [isDownloading, setIsDownloading] = useState(false);
+
+  const handleDownload = async () => {
+    setIsDownloading(true);
+    const formattedDate = formatUnixTimestampToDateTime(startTime, {
+      strftime_fmt: "%D-%T",
+      time_style: "medium",
+      date_style: "medium",
+    });
+    const filename = `${camera}_${formattedDate}.mp4`;
+
+    try {
+      const response = await fetch(source);
+      const blob = await response.blob();
```

Comment:
@NickM-27 is this essentially reading the whole MP4 file from the server into the browser's memory, then creating a dummy link to click it to trigger the "download"? 

If so, that feels like a bit of a risk e.g. for low memory phones. 
https://chromium.googlesource.com/chromium/src/+/master/storage/browser/blob/README.md#blob-storage-limits also seems to suggest that on Android, Chrome might page blobs to disk at 20MB. A 4K, 25fps camera at a bitrate of 16mbps is ~ 60MB for a 30 second file.

Reply:
A direct download link is probably preferable. I'll put up another PR later.

Reply:
https://github.com/blakeblackshear/frigate/pull/14347